iphone/Classes/KrollContext.m

```diff
@@ -1177,10 +1179,15 @@ - (void)main
   [kroll release];
   TiStringRelease(prop);
 
+#if defined(USE_JSCORE_FRAMEWORK) && !defined(TI_USE_KROLL_THREAD)
```

Comment:
I think the problem is with JSCore only irrespective of main thread or kroll thread. If so we should update the # if defined statement to - 
#if defined(USE_JSCORE_FRAMEWORK)